File: server/feed/src/feed/cassandra.clj
(ns feed.cassandra)

(require '[qbits.alia :as alia])
(require '[feed.settings :as prop])

(def cluster (if (nil? prop/service-config) nil (alia/cluster (:nosql-host prop/service-config))))
(def session (if (nil? prop/service-config) nil (alia/connect cluster "activity")))

(if (not (nil? prop/service-config)) (alia/set-consistency! (keyword (:nosql-consistency prop/service-config))))

(def load-inbound-from-db-command "select dateOf(occurred), fromparticipantid, subject, story from Inbound where participantid = ? order by occurred desc")
(def save-inbound-to-db-command (str "insert into Inbound (ParticipantID, FromParticipantID, Occurred, Subject, Story) values (?, ?, now(), ?, ?) using ttl " (:nosql-ttl prop/service-config)))
(def load-outbound-from-db-command "select dateOf(occurred), subject, story from Outbound where participantid = ? order by occurred desc")
(def save-outbound-to-db-command "insert into Outbound (ParticipantID, Occurred, Subject, Story) values (?, now(), ?, ?)")

File: server/feed9/go/cassandra.go
package newsfeedserver

import (
        "os"
	"log"
	"time"
	"github.com/gocql/gocql"
)

func CreateSession()(*gocql.Session) {
	cluster := gocql.NewCluster(os.Getenv("NOSQL_HOST"))
	cluster.Keyspace = os.Getenv("NOSQL_KEYSPACE")
	cluster.Timeout = 10 * time.Second
	cluster.ConnectTimeout = 20 * time.Second
	cluster.Consistency = gocql.Any
	retVal, err := cluster.CreateSession()
	if err != nil {
	   log.Println(err)
	}
	return retVal
}

var session = CreateSession()

type AddCassandraWrapper interface {
        AddInbound(i Inbound)
        AddOutbound(o Outbound)
}

type CassandraWrapper struct {
	Session *gocql.Session
}

func (cw CassandraWrapper) AddOutbound(o Outbound) {
	from, err := ExtractId(o.From)
	if err != nil {
	   return
	}
	stmt := cw.Session.Query("insert into Outbound (ParticipantID, Occurred, Subject, Story) values (?, now(), ?, ?) using ttl 7776000", from, o.Subject, o.Story)
	stmt.Consistency(gocql.One)
	stmt.Exec()
}

func (cw CassandraWrapper) AddInbound(i Inbound) {
	from, err := ExtractId(i.From)
	if err != nil {
	   return
	}
	to, err := ExtractId(i.To)
	if err != nil {
	   return
	}
	stmt := cw.Session.Query("insert into Inbound (ParticipantID, FromParticipantID, Occurred, Subject, Story) values (?, ?, now(), ?, ?) using ttl 7776000", to, from, i.Subject, i.Story)
	stmt.Consistency(gocql.Any)
	stmt.Exec()
}

func connectCassandra() (*CassandraWrapper) {
	retVal := CassandraWrapper{
	   Session: session,
	}
	return &retVal
}